core/services/relay/evm/evmtesting/run_tests.go

```diff
@@ -33,194 +43,242 @@ func RunChainComponentsInLoopEvmTests[T TestingT[T]](t T, it ChainComponentsInte
-
-		assert.Equal(t, &anyString, rOutput.FieldByName("Field").Interface())
-		topic, err := abi.MakeTopics([]any{anyString})
-		require.NoError(t, err)
-		assert.Equal(t, &topic[0][0], rOutput.FieldByName("FieldHash").Interface())
-	})
-
-	t.Run("Multiple topics can filter together", func(t T) {
-		it.Setup(t)
-		ctx := it.Helper.Context(t)
-		cr := it.GetContractReader(t)
-		bindings := it.GetBindings(t)
-
-		require.NoError(t, cr.Bind(ctx, bindings))
-
-		triggerFourTopics(t, it, int32(1), int32(2), int32(3))
-		triggerFourTopics(t, it, int32(2), int32(2), int32(3))
-		triggerFourTopics(t, it, int32(1), int32(3), int32(3))
-		triggerFourTopics(t, it, int32(1), int32(2), int32(4))
-
-		var bound types.BoundContract
-		for idx := range bindings {
-			if bindings[idx].Name == AnyContractName {
-				bound = bindings[idx]
-			}
-		}
-
-		var latest struct{ Field1, Field2, Field3 int32 }
-		params := struct{ Field1, Field2, Field3 int32 }{Field1: 1, Field2: 2, Field3: 3}
-
-		time.Sleep(it.MaxWaitTimeForEvents())
-
-		require.NoError(t, cr.GetLatestValue(ctx, bound.ReadIdentifier(triggerWithAllTopics), primitives.Unconfirmed, params, &latest))
-		assert.Equal(t, int32(1), latest.Field1)
-		assert.Equal(t, int32(2), latest.Field2)
-		assert.Equal(t, int32(3), latest.Field3)
-	})
-
-	t.Run("Filtering can be done on indexed topics that get hashed", func(t T) {
-		it.Setup(t)
-
-		cr := it.GetContractReader(t)
-		ctx := it.Helper.Context(t)
-		bindings := it.GetBindings(t)
-
-		require.NoError(t, cr.Bind(ctx, bindings))
-
-		triggerFourTopicsWithHashed(t, it, "1", [32]uint8{2}, [32]byte{5})
-		triggerFourTopicsWithHashed(t, it, "2", [32]uint8{2}, [32]byte{3})
-		triggerFourTopicsWithHashed(t, it, "1", [32]uint8{3}, [32]byte{3})
-
-		var bound types.BoundContract
-		for idx := range bindings {
-			if bindings[idx].Name == AnyContractName {
-				bound = bindings[idx]
-			}
-		}
-
-		var latest struct {
-			Field3 [32]byte
-		}
-		params := struct {
-			Field1 string
-			Field2 [32]uint8
-			Field3 [32]byte
-		}{Field1: "1", Field2: [32]uint8{2}, Field3: [32]byte{5}}
-
-		time.Sleep(it.MaxWaitTimeForEvents())
-		require.NoError(t, cr.GetLatestValue(ctx, bound.ReadIdentifier(triggerWithAllTopicsWithHashed), primitives.Unconfirmed, params, &latest))
-		// only checking Field3 topic makes sense since it isn't hashed, to check other fields we'd have to replicate solidity encoding and hashing
-		assert.Equal(t, [32]uint8{5}, latest.Field3)
-	})
-
-	t.Run("Bind returns error on missing contract at address", func(t T) {
-		it.Setup(t)
-
-		addr := common.BigToAddress(big.NewInt(42))
-		reader := it.GetContractReader(t)
-
-		ctx := it.Helper.Context(t)
-		err := reader.Bind(ctx, []clcommontypes.BoundContract{{Name: AnyContractName, Address: addr.Hex()}})
-
-		require.ErrorIs(t, err, read.NoContractExistsError{Address: addr})
-	})
+	testCases := []Testcase[T]{
+		{
+			Name: DynamicTypedTopicsFilterAndCorrectReturn,
```

Comment:
Is this leakage from [BCFR-967?](https://github.com/smartcontractkit/chainlink-common/pull/829) ?